internal/inputs/inputs.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inputs
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/google/gke-policy-automation/internal/log"
+)
+
+const (
+	defaultMaxDataGetCoroutines = 20
+)
+
+type Input interface {
+	GetID() string
+	GetDescription() string
+	GetData(clusterID string) (interface{}, error)
+	Close() error
+}
+
+type Cluster struct {
+	Data map[string]interface{}
+}
+
+type getDataTask struct {
+	input     Input
+	clusterID string
+}
+
+type getDataTaskResult struct {
+	clusterID string
+	inputID   string
+	result    interface{}
+	err       error
+}
+
+//GetAllInputsData fetches data from given inputs for all given clusters in a concurrent manner
+func GetAllInputsData(inputs []Input, clusterIDs []string) (map[string]*Cluster, []error) {
+	tasksNo := len(inputs) * len(clusterIDs)
+	tasksChan := make(chan *getDataTask, tasksNo)
```

Comment:
This channel does not have to be that big. A size of `defaultMaxDataGetCoroutines` would be optimal if we fill it in separate goroutine.

Reply:
Good point, thank you!